app/models/katello/glue/provider.rb

```diff
@@ -34,7 +34,7 @@ module InstanceMethods
       API_URL = 'https://subscription.rhsm.redhat.com/subscription/consumers/'.freeze
       def api_url(upstream = {})
         # Default to Red Hat
-        upstream['apiUrl'] || API_URL
+        upstream['apiUrl'] || ENV['REDHAT_RHSM_API_URL'] || API_URL
```

Comment:
This means that you won't be able to control this via environment variable if the upstream metadata already contains `apiUrl`. Is this what we want?

if not then maybe should switch the order

Reply:
hmm good point :)

```suggestion
        ENV['REDHAT_RHSM_API_URL'] || upstream['apiUrl'] || API_URL
```